# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/android/DocAccessibleWrap.cpp
# Commit: 4965ca9b7882
# Full Hash: 4965ca9b7882e482c0d5f48de63fad4457a9863f
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2022-07-07 09:37:57
# Regressor Bug: 1771934
# File Overlap Count: 1
# Description:
#   Bug 1771934 - Unregister all SessionAccessibility accessibles when top doc is shut down. r=Jamie
#   
#   The associated PresShell of the root doc can be used for retrieving the
#   SessionAccessibility. If the PresShell is about to go away, we should
#   unregister all the accessibles.
# ==============================================================================

diff -r 3827c37b916d -r 4965ca9b7882 accessible/android/DocAccessibleWrap.cpp
--- a/accessible/android/DocAccessibleWrap.cpp	Thu Jul 07 01:47:08 2022 +0300
+++ b/accessible/android/DocAccessibleWrap.cpp	Wed Jul 06 22:26:08 2022 +0000
@@ -53,7 +53,11 @@
   // Unregister here before disconnecting from PresShell.
   if (!IPCAccessibilityActive()) {
     MonitorAutoLock mal(nsAccessibilityService::GetAndroidMonitor());
-    SessionAccessibility::UnregisterAccessible(this);
+    if (IsRoot()) {
+      SessionAccessibility::UnregisterAll(PresShellPtr());
+    } else {
+      SessionAccessibility::UnregisterAccessible(this);
+    }
   }
   DocAccessible::Shutdown();
 }